tar/bsdtar.c

```diff
@@ -783,14 +790,15 @@ main(int argc, char **argv)
 	 * few bytes off anyway since the command line, including "--dry-run"
 	 * is included in the metadata.
 	 */
-	if (bsdtar->option_dryrun && (bsdtar->ntapes == 0)) {
+	if (bsdtar->option_dryrun &&
+	    (strlist_getsize(bsdtar->tapenames_setup) == 0)) {
```

Comment:
This is a bit ugly, but if we're not incrementing `bsdtar->ntapes`, I guess we don't have an alternative.